modules/nf-core/propr/propr/templates/propr.R

```diff
@@ -96,6 +96,79 @@ set_reference <- function(ivar, mat){
     return(ivar)
 }
 
+#' Set the appropiate range for the sequence of cutoffs used in updateCutoffs.
+#' Adjusts the interval to the different metrics.
+#'
+#' @param object propr object. Output from propr function.
+#'
+#' @return sequence of cutoff values.
+seqCutoff <- function(object){
+    matrix <- getMatrix(object)
+    matrix[matrix <= 0] <- NA
+    diag(matrix) <- NA
+    min_cutoff <- round(min(matrix, na.rm = TRUE),3)
+    max_cutoff <- round(max(matrix, na.rm = TRUE),3)
+    step_cutoff <- (max_cutoff - min_cutoff)/ 20
+    seq_cutoff <- seq(min_cutoff, max_cutoff, step_cutoff)
+    return(seq_cutoff)
+}
+
+#' Extract the proportionality cutoff for a specified FDR value.
+#' Gene pairs with a proportionality value higher than the extracted cutoff will be considered significantly proportional.
+#'
+#' @param object propr object. Output from propr function. updateCutoffs function should be applied to the object previous to valCutoff.
+#' @param fdrVal FDR value to extract the cutoff for. Per default 0.05
+#' @param metric Metric used to calculate the proportionality values. Options are 'cor', 'rho', 'phi', 'phs', 'vlr', 'pcor', 'pcor.shrink', 'pcor.bshrink'
+#'
+#' @return cutoff value. Proportionality values higher than this cutoff are considered significant.
+valCutoff  <- function(object, metric, fdrVal = 0.05){
+    fdr_df <- object@fdr
+    print(fdr_df)
+    metric_up <- c("rho", "cor", "pcor", "pcor.shrink", "pcor.bshrink")
+    if (prod(dim(fdr_df) == 0)){
+        warning("Please run updateCutoff on propr first")
+    }else{
+        fdr_vals <- fdr_df\$FDR
+        if (any(is.na(fdr_vals))){
+            stop("FDR not defined. This metric is not appropiate for the given dataset")
+        }
+        threshold <- any(fdr_vals <= fdrVal)
+        if (metric %in% metric_up){
+            if (threshold){
+                fdr_threshold <- fdr_vals[which.max(fdr_vals <= fdrVal)]
```

Comment:
`fdr_vals <= fdrVal` returns a vector of boolean values no?
Does `which.max` or `which.min` work well with boolean values?